odbc_fdw.c

```diff
@@ -1367,3 +1578,288 @@ odbcReScanForeignScan(ForeignScanState *node)
+				 NULL, 0, /* Table */
+				 (SQLCHAR*)"TABLE", SQL_NTS /* Type of table (we're not interested in views, temporary tables, etc.) */
+			);
+			check_return(ret, "Obtaining ODBC tables", tables_stmt, SQL_HANDLE_STMT);
+
+			initStringInfo(&col_str);
+			SQLCHAR *table_catalog = (SQLCHAR *) palloc(sizeof(SQLCHAR) * 255);
+			SQLCHAR *table_schema = (SQLCHAR *) palloc(sizeof(SQLCHAR) * 255);
+			while (SQL_SUCCESS == ret)
+			{
+				ret = SQLFetch(tables_stmt);
+				if (SQL_SUCCESS == ret)
+				{
+					int excluded = FALSE;
+					TableName = (SQLCHAR *) palloc(sizeof(SQLCHAR) * 255);
+					ret = SQLGetData(tables_stmt, 3, SQL_C_CHAR, TableName, 255, &indicator);
+					check_return(ret, "Reading table name", tables_stmt, SQL_HANDLE_STMT);
+
+					/* Since we're not filtering the SQLTables call by schema
+					   we must exclude here tables that belong to other schemas.
+					   For some ODBC drivers tables may not be organized into
+					   schemas and the schema of the table will be blank.
+					   So we only reject tables for which the schema is not
+					   blank and different from the desired schema:
+					 */
+					ret = SQLGetData(tables_stmt, 2, SQL_C_CHAR, table_schema, 255, &indicator);
+ 					if (table_schema[0] != 0 && strcmp(table_schema, schema_name) )
+ 					{
+ 						excluded = TRUE;
+ 					}
+
+					/* Since we haven't specified SQL_ALL_CATALOGS in the
+					   call to SQLTables we shouldn't get tables from special
+					   catalogs and only from the regular catalog of the database
+					   (named as the database or blank, depending on the driver)
+					   but to be sure we'll reject tables from catalogs with
+					   other names:
+					 */
+					ret = SQLGetData(tables_stmt, 1, SQL_C_CHAR, table_catalog, 255, &indicator);
+					if (table_catalog[0] != 0 && strcmp(table_catalog, schema_name))
+					{
+						if (options.database == NULL || strcmp(table_catalog, options.database))
+						{
+							excluded = TRUE;
+						}
+					}
+
+					/* And now we'll handle tables excluded by an EXCEPT clause */
+					if (!excluded && stmt->list_type == FDW_IMPORT_SCHEMA_EXCEPT)
+					{
+						foreach(tables_cell,  stmt->table_list)
+						{
+							table_rangevar = (RangeVar*)lfirst(tables_cell);
+							if (strcmp(TableName, table_rangevar->relname) == 0)
+							{
+								excluded = TRUE;
+							}
+						}
+					}
+
+					if (!excluded)
+					{
+						tables = lappend(tables, (void*)TableName);
+					}
+				}
+			}
+			SQLCloseCursor(tables_stmt);
+			SQLFreeHandle(SQL_HANDLE_STMT, tables_stmt);
+		}
+		else if (stmt->list_type == FDW_IMPORT_SCHEMA_LIMIT_TO)
+		{
+			foreach(tables_cell, stmt->table_list)
+			{
+				table_rangevar = (RangeVar*)lfirst(tables_cell);
+				tables = lappend(tables, (void*)table_rangevar->relname);
+			}
+		}
+		else
+		{
+			elog(ERROR,"Unknown list type in IMPORT FOREIGN SCHEMA");
+		}
+        foreach(tables_cell, tables)
```

Comment:
We're using tabulators as specified in PG style and github expands them to 8-char. boundaries instead of 4.